mixer/pkg/log/log_test.go

```diff
@@ -0,0 +1,233 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"errors"
+	"io/ioutil"
+	"log"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"google.golang.org/grpc/grpclog"
+)
+
+func TestBasic(t *testing.T) {
+	cases := []struct {
+		f          func()
+		pat        string
+		json       bool
+		caller     bool
+		stackLevel zapcore.Level
+	}{
+		{func() { Debug("Hello") }, ".*Z\tdebug\tHello", false, false, None},
+		{func() { Debugf("Hello") }, ".*Z\tdebug\tHello", false, false, None},
+		{func() { Debugw("Hello") }, ".*Z\tdebug\tHello", false, false, None},
+		{func() { Debuga("Hello") }, ".*Z\tdebug\tHello", false, false, None},
+
+		{func() { Info("Hello") }, ".*Z\tinfo\tHello", false, false, None},
+		{func() { Infof("Hello") }, ".*Z\tinfo\tHello", false, false, None},
+		{func() { Infow("Hello") }, ".*Z\tinfo\tHello", false, false, None},
+		{func() { Infoa("Hello") }, ".*Z\tinfo\tHello", false, false, None},
+
+		{func() { Warn("Hello") }, ".*Z\twarn\tHello", false, false, None},
+		{func() { Warnf("Hello") }, ".*Z\twarn\tHello", false, false, None},
+		{func() { Warnw("Hello") }, ".*Z\twarn\tHello", false, false, None},
+		{func() { Warna("Hello") }, ".*Z\twarn\tHello", false, false, None},
+
+		{func() { Error("Hello") }, ".*Z\terror\tHello", false, false, None},
+		{func() { Errorf("Hello") }, ".*Z\terror\tHello", false, false, None},
+		{func() { Errorw("Hello") }, ".*Z\terror\tHello", false, false, None},
+		{func() { Errora("Hello") }, ".*Z\terror\tHello", false, false, None},
+
+		{func() {
+			l := With(zap.String("key", "value"))
+			l.Debug("Hello")
+		}, ".*Z\tdebug\tHello\t{\"key\": \"value\"}", false, false, None},
+
+		{func() { Debug("Hello") }, ".*Z\tdebug\tlog/log_test.go:.*\tHello", false, true, None},
+
+		{func() { Debug("Hello") }, "{\"level\":\"debug\",\"time\":\".*T.*Z\",\"caller\":\"log/log_test.go:.*\",\"msg\":\"Hello\",\"stack\":\".*\"}",
+			true, true, zapcore.DebugLevel},
+		{func() { Info("Hello") }, "{\"level\":\"info\",\"time\":\".*T.*Z\",\"caller\":\"log/log_test.go:.*\",\"msg\":\"Hello\",\"stack\":\".*\"}",
+			true, true, zapcore.DebugLevel},
+		{func() { Warn("Hello") }, "{\"level\":\"warn\",\"time\":\".*T.*Z\",\"caller\":\"log/log_test.go:.*\",\"msg\":\"Hello\",\"stack\":\".*\"}",
+			true, true, zapcore.DebugLevel},
+		{func() { Error("Hello") }, "{\"level\":\"error\",\"time\":\".*T.*Z\",\"caller\":\"log/log_test.go:.*\",\"msg\":\"Hello\",\"stack\":\".*\"}",
+			true, true, zapcore.DebugLevel},
+	}
+
+	for i, c := range cases {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			lines, err := captureStdout(func() {
```

Comment:
Does this mean the output will also be printed in the console when executing the test ?

Reply:
Yes, but the test harness eats it up. The only time the stdout output is displayed is when a test fails.